web3/_utils/http_session_manager.py

```diff
@@ -80,7 +86,7 @@ def cache_and_return_session(
                     f"{evicted_session}",
                 )
             threading.Timer(
-                DEFAULT_TIMEOUT + 0.1,
+                request_timeout + 0.1,
```

Comment:
If this is None like the type allows, I think this line will throw?  Maybe add a `if request_timeout is None` check? But I'm also surprised mypy didn't catch that so maybe I missed something.

Reply:
Good eye... I'll make sure to address it.